master/internal/command/command.go

```diff
@@ -206,8 +207,16 @@ func (c *command) Receive(ctx *actor.Context) error {
 		case msg.Container.State == container.Running:
 			c.addresses = msg.ContainerStarted.Addresses
 
-			names := make([]string, 0, len(c.addresses))
+			var names []string
 			for _, address := range c.addresses {
+				// Only proxy the port we expect to proxy.  If a dockerfile uses an EXPOSE command,
+				// additional addresses will appear her, but currently we only proxy one uuid to one
+				// port, so it doesn't make sense to send multiple proxy.Register messages for a
+				// single ServiceID (only the last one would work).
+				if c.assignedPort == 0 || address.ContainerPort != c.assignedPort {
```

Comment:
this pattern of retrieving the ports we wanted mapped from the container addresses is the exact one used in rendezvous. with trying to bring commands and trials closer, what do you think about extracting this into a helper
`AddressesInRange(start, end int, addresses ...[]sproto.Address)`. since it would conflict my rendezvous work, there is no reason to do it here. but maybe drop a TODO and ticket so it's not forgotten.

Reply:
https://determinedai.atlassian.net/browse/DET-5682